apps/sms/views/inbox/js/inbox.js

```diff
@@ -914,107 +794,57 @@ var InboxView = {
-    for (var i = 0, l = threads.length; i < l; i++) {
-      if (timestamp > threads[i].dataset.time) {
-        threadFound = true;
-        threadsContainer.insertBefore(node, threads[i]);
-        break;
+    return this.createThread(thread).then((data) => {
+      // Async update contact after insert to list
+      this.setContact(data);
+
+      // Where have I to place the new thread?
+      var threadFound = false;
+      for (var i = this.listData.length - 1; i > -1 ; i--) {
+        if (timestamp < this.listData[i].timestamp) {
+          threadFound = true;
+          this.listData.splice(i + 1, 0, data);
+          break;
+        }
       }
-    }
-
-    if (!threadFound) {
-      threadsContainer.appendChild(node);
-    }
 
-    if (this.inEditMode) {
-      // Remove the new added thread id from the selection handler
-      this.selectionHandler.unselect(thread.id);
+      if (!threadFound) {
+        this.listData.unshift(data);
+      }
 
-      this.updateSelectionStatus();
-    }
+      if (this.inEditMode) {
+        // Remove the new added thread id from the selection handler
+        this.selectionHandler.unselect(thread.id);
 
-    return firstThreadInContainer;
-  },
+        this.updateSelectionStatus();
+      }
 
-  // Adds a new grouping header if necessary (today, tomorrow, ...)
-  createThreadContainer: function inbox_createThreadContainer(timestamp) {
-    var threadContainer = document.createElement('div');
-    // Create Header DOM Element
-    var headerDOM = document.createElement('header');
-
-    // The id is used by the sticky header code as the -moz-element target.
-    headerDOM.id = 'header_' + timestamp;
-
-    // Append 'time-update' state
-    headerDOM.dataset.timeUpdate = 'repeat';
-    headerDOM.dataset.time = timestamp;
-    headerDOM.dataset.dateOnly = true;
-
-    // Create UL DOM Element
-    var threadsContainerDOM = document.createElement('ul');
-    threadsContainerDOM.id = 'threadsContainer_' +
-                              Utils.getDayDate(timestamp);
-    // Add text
-    Utils.setHeaderDate({
-      time: timestamp,
-      element: headerDOM
+      return;
     });
-
-    // Add to DOM all elements
-    threadContainer.appendChild(headerDOM);
-    threadContainer.appendChild(threadsContainerDOM);
-    return threadContainer;
   },
 
   // Method for updating all contact info after creating a contact
   updateContactsInfo: function inbox_updateContactsInfo() {
     Contacts.clearUnknown();
     // Prevents cases where updateContactsInfo method is called
-    // before InboxView.container exists (as observed by errors
-    // in the js console)
-    if (!this.container) {
+    // before listData exists.
+    if (!this.listData) {
       return;
     }
-    // Retrieve all 'li' elements
-    var threads = this.container.getElementsByTagName('li');
-
-    [].forEach.call(threads, this.setContact.bind(this));
+    // Retrieve all the list data
+    this.listData.forEach(this.setContact);
```

Comment:
It looks like after contact change we update model items, but list re-renders it after some time only when you try to scroll up and down :)